samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/delivery/DeliveryBean.java

```diff
@@ -2547,21 +2556,19 @@ public String addMediaToItemGrading(String mediaLocation)
 
     // 2. format of the media location is: assessmentXXX/questionXXX/agentId/myfile
     // 3. get the questionId (which is the PublishedItemData.itemId)
-    //int assessmentIndex = mediaLocation.indexOf("assessment");
-    int questionIndex = mediaLocation.indexOf("question");
-    int agentIndex = mediaLocation.indexOf("/", questionIndex + 8);
-    int myfileIndex = mediaLocation.lastIndexOf("/");
-    //cwen
-    if(agentIndex < 0 )
-    {
-      agentIndex = mediaLocation.indexOf("\\", questionIndex + 8);
-    }
-    //String pubAssessmentId = mediaLocation.substring(assessmentIndex + 10, questionIndex - 1);
-    String questionId = mediaLocation.substring(questionIndex + 8, agentIndex);
+    String systemMediaLocation = mediaLocation.replace(File.separator.equals("/") ? "\\" : "/" , File.separator);
+    String[] mediaLocationParts = systemMediaLocation.split(Pattern.quote(File.separator));
+    int mediaLocationTokens = mediaLocationParts.length;
+    String fileName = mediaLocationParts[mediaLocationTokens- 1];
```

Comment:
I'd change mediaLocationTokens to numberOfParts, something like that.

Reply:
I'd also put a sanity check in for the number of parts, to make sure your index doesn't go negative.